### PR TITLE
fix: Support ImageRef in WindowsProfile

### PR DESCRIPTION
**Pull request by AbelHu**
**Reason for Change**:

Support ImageRef and CustomImage in WindowsProfile.

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [ ] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Review by mainred — APPROVED** [review]
LGTM, but 
1. we need @xuto2 ' approval.
2. Considering the complicated logic and string concatenation, I'd suggest we have some test to verify this change and for future protection.
3. We have similar logic for vmss and vm, we can scope the common operations into a function. But considering the future support of only vmss(Please correct me if I am wrong), we can keep the code like this.

File: pkg/engine/virtualmachines.go
Comment (xuto2) [inline pkg/engine/virtualmachines.go:506]: the rule is a bit complicated, why do we need to use both agent pool and windows profile? can we not combine them into one?

File: pkg/engine/params.go
Comment (xuto2) [inline pkg/engine/params.go:229]: why do we have to use this? Is agentProfile.ImageRef not enough?

File: pkg/engine/virtualmachines.go
Comment (AbelHu) [reply pkg/engine/virtualmachines.go:506]: Current code is using ImageRef in WindowsProfile. To support more Windows image, for example, 2004, we need to support multiple VHD images in the same cluster so we need to use ImageRef (SIG uses) in agent pool. Later the VHD info in WindowsProfile will be deprecated.

File: pkg/engine/params.go
Comment (AbelHu) [reply pkg/engine/params.go:229]: I found that current code in AKS is using WindowsProfile.ImageRef. If some regions are using this, it will fail. I need the confirmation from @xizhamsft

Comment (AbelHu) [reply pkg/engine/params.go:229]: @xuto2 WindowsProfile.ImageRef is used by CustomizedWindowsOSImage.
After syncing with @xizhamsft, WindowsProfile.ImageRef is not used by those regions for now. I will hold this PR.

**Comment by AbelHu** [comment]
@mainred I used the SIG you provided to test customized sig and the test passed.

**Pull request merged**
